.github/workflows/azure_e2e_release_promotion_tests.yml

```diff
@@ -0,0 +1,190 @@
+        id: gen
+        run: |
+          project_name=$(echo "tmp-billing-project-$(uuidgen)" | cut -c -30)
+          echo "project_name=${project_name}" >> $GITHUB_OUTPUT
+  create-bee-workflow:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - name: Echo Leo version
+        run: >
+          echo '${{ needs.leo-build-tag-publish-job.outputs.custom-version-json
+          }}'
+      - name: dispatch to terra-github-workflows
+        uses: broadinstitute/workflow-dispatch@v4.0.0
+        with:
+          run-name: '${{ env.BEE_CREATE_RUN_NAME }}'
+          workflow: bee-create
+          repo: broadinstitute/terra-github-workflows
+          ref: refs/heads/main
+          token: '${{ env.TOKEN }}'
+          inputs: >-
+            { "run-name": "${{ env.BEE_CREATE_RUN_NAME }}", "bee-name": "${{
+            env.BEE_NAME }}", "bee-template-name": "rawls-e2e-azure-tests", 
+            "version-template": "staging" }
+  attach-landing-zone-to-bee-workflow:
+    runs-on: ubuntu-latest
+    needs:
+      - init-github-context
+      - create-bee-workflow
+      - params-gen
+    steps:
+      - name: dispatch to terra-github-workflows
+        uses: broadinstitute/workflow-dispatch@v4.0.0
+        with:
+          run-name: '${{ env.ATTACH_BP_TO_LZ_RUN_NAME }}'
+          workflow: attach-billing-project-to-landing-zone.yaml
+          repo: broadinstitute/terra-github-workflows
+          ref: refs/heads/main
+          token: '${{ env.TOKEN }}'
+          inputs: >-
+            { "run-name": "${{ env.ATTACH_BP_TO_LZ_RUN_NAME }}", "bee-name": "${{
+            env.BEE_NAME }}", "billing-project": "${{
+            needs.params-gen.outputs.project-name }}"}
+  create-sam-pet-job:
+    permissions:
+      contents: read
+      id-token: write
+    needs:
+      - attach-landing-zone-to-bee-workflow
+      - params-gen
+    runs-on: ubuntu-latest
+    # We need to create the pet before starting the tests, otherwise all 3 will do so at the same time and interfere with each other
+    steps:
+      - name: Generate OAuth2 2.0 access token for owner
+        id: 'obtain_token'
+        uses: google-github-actions/auth@v1
+        with:
+          token_format: 'access_token'
+          workload_identity_provider: 'projects/1038484894585/locations/global/workloadIdentityPools/github-wi-pool/providers/github-wi-provider'
+          service_account: 'firecloud-qa@broad-dsde-qa.iam.gserviceaccount.com'
+          access_token_scopes: 'profile, email, openid'
+          access_token_subject: 'hermione.owner@quality.firecloud.org'
+          export_environment_variables: false
+
+      - name: Call Sam to create pet
+        run: |
+          SAM_URL="https://sam.${{ env.BEE_NAME }}.bee.envs-terra.bio"
+          API_URL="${SAM_URL}/api/azure/v1/user/petManagedIdentity"
+          AZURE_TOKEN="${{ steps.obtain_token.outputs.access_token }}"
+          PAYLOAD='{
```

Comment:
These are the default values that I see used in the create bee/attach bp workflows.  I put them in directly here just to get this step working; should they be environment variables at the beginning of the yaml file?  Should there be an option to pass them in and then pass them along to the other workflows?